aten/src/ATen/native/cuda/MemoryAccess.cuh

```diff
@@ -486,7 +486,9 @@ inline C10_HOST_DEVICE int can_vectorize_up_to(const char *pointer) {
   uint64_t address = reinterpret_cast<uint64_t>(pointer);
   constexpr int vec2_alignment = std::alignment_of_v<aligned_vector<scalar_t, 2>>;
   constexpr int vec4_alignment = std::alignment_of_v<aligned_vector<scalar_t, 4>>;
+#if defined(USE_ROCM) || (defined(CUDA_VERSION) && CUDA_VERSION >= 12080)
```

Comment:
should `USE_ROCM` here also be inverted if the `CUDA_VERSION` condition is `>= 12080`

Reply:
No, I don't think so. Before https://github.com/pytorch/pytorch/pull/145746 vec8_alignment were only available to `USE_ROCM`, after it was enabled unconditionally and I want it to be enabled for either ROCM or CUDA newer than 12.6